recipes/icu/all/conanfile.py

```diff
@@ -368,6 +368,10 @@ def package_info(self):
             self.output.info("Appending PATH environment variable: {}".format(bin_path))
             self.env_info.PATH.append(bin_path)
 
+        if self.options.shared:
+            self.env_info.LD_LIBRARY_PATH.append(os.path.join(self.package_folder, "lib"))
+            self.env_info.DYLD_LIBRARY_PATH.append(os.path.join(self.package_folder, "lib"))
+
```

Comment:
What? No, it's not the job of icu recipe to manage LD_LIBRARY_PATH or DYLD_LIBRARY_PATH for consumers.

Reply:
@SpaceIm shared icu library include symlink like `libicudata.so.71`, without defining LD_LIBRARY_PATH in icu recipe, all consumer has to set LD_LIBRARY_PATH to icu lib dir. That is an unnecessary inconvenience I guess.

Reply:
so LD_LIBRARY_PATH/DYLD_LIBRARY_PATH is not the solution, because they only have an impact at run-time, not at link-time.  and the issue is not the same as #11388 which mentionned no link failure. Please make a full issue report at https://github.com/conan-io/conan-center-index/issues/new?assignees=&labels=bug&template=package_bug.yml&title=%5Bpackage%5D+%3CLIBRARY-NAME%3E%2F%3CLIBRARY-VERSION%3E%3A+SHORT+DESCRIPTION